src/org/intellij/erlang/inspection/ErlangUnresolvedFunctionInspection.java

```diff
@@ -39,27 +46,25 @@ public void visitFunctionCallExpression(@NotNull ErlangFunctionCallExpression o)
         PsiReference reference = o.getReference();
         if (reference instanceof ErlangFunctionReference && reference.resolve() == null) {
           if (o.getQAtom().getMacros() != null) return;
-          ErlangFunctionReference r = (ErlangFunctionReference) reference;
-
-          String name = r.getName();
-          int arity = r.getArity();
-
-          if (arity < 0) return;
 
-          String signature = r.getSignature();
+          int arity = o.getArgumentList().getExpressionList().size();
+          String name = o.getName();
+          String functionPresentation = ErlangPsiImplUtil.createFunctionPresentation(name, arity);
 
           PsiElement parent = o.getParent();
           if (parent instanceof ErlangGlobalFunctionCallExpression) {
             ErlangModuleRef moduleRef = ((ErlangGlobalFunctionCallExpression) parent).getModuleRef();
             if (moduleRef.getQAtom().getMacros() != null) return;
-            signature = moduleRef.getText() + ":" + signature;
+            functionPresentation = moduleRef.getText() + ":" + functionPresentation;
           }
+          String fixMessage = "Create Function " + functionPresentation;
 
-          LocalQuickFix[] qfs = parent instanceof ErlangGenericFunctionCallExpression || parent instanceof ErlangGlobalFunctionCallExpression ?
-            new LocalQuickFix[]{} :
-            new LocalQuickFix[]{new ErlangCreateFunctionQuickFix(name, arity)};
+          LocalQuickFix[] fixes = parent instanceof ErlangGenericFunctionCallExpression ||
+                                  parent instanceof ErlangGlobalFunctionCallExpression ? new LocalQuickFix[0] :
+                                  new LocalQuickFix[]{new ErlangCreateFunctionQuickFix(createProviderForFunctionCall(o),
+                                                                                       fixMessage)};
```

Comment:
You can replace your ```createProviderForFunctionCall``` method with the one that returns a CreateFunctionQuickFix - it will be more readable.